File: QytetetRuby/pkg/RubyQytetet-0.0.1/lib/titulo_propiedad.rb
#encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module ModeloQytetet
  class TituloPropiedad
    
    attr_accessor :hipotecado
    attr_reader :calle, :alquilerBase, :factorRevaloracion, :hipotecaBase, :precioEdificar
    attr_accessor :casilla, :propietario
    
    def initialize(c,ab,fr,hb,pe,propie,casi)
      
      @calle = c
      @hipotecado = false
      @alquilerBase = ab
      @factorRevaloracion = fr
      @hipotecaBase = hb
      @precioEdificar = pe
      @propietario=propie
      @casilla=casi
      
    end
    
    def self.inicializar(c,ab,fr,hb,pe)
      
      self.new(c,ab,fr,hb,pe,nil,nil)
    end
    
    def to_s
      "Calle: #{@calle} \n Hipotecado: #{@hipotecado} \n AlquilerBase: #{@alquilerBase} \n FactorRevaloracion: #{@factorRevaloracion} \n HipotecaBase: #{@hipotecaBase} \n PrecioEdificar: #{@precioEdificar}"
    end
    
    def cobrar_alquiler(coste)
      
    end
    
    def propietario_encarcelado()
      
    end
    
    def set_casilla(casilla)
      
    end
    
    def set_propietario(propietario)
      
    end
    
    def tengo_propietario() 
      
    end

  end
end

File: QytetetJava/src/modeloqytetet/TituloPropiedad.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

/**
 *
 * 
 */
public class TituloPropiedad {
  private Casilla casilla;
  private Jugador propietario;
  private String nombre;
  private Boolean hipotecado = false;
  private int alquilerBase;
  private double factorRevaloracion;
  private int hipotecaBase;
  private int precioEdificar;
  
  TituloPropiedad (String nombre, int alquilerBase, double factorRevaloracion, int hipotecaBase, int precioEdificar) {
      this.nombre = nombre;
      this.alquilerBase = alquilerBase;
      this.factorRevaloracion = factorRevaloracion;
      this.hipotecaBase = hipotecaBase;
      this.precioEdificar = precioEdificar;
  }
    
    Casilla getCasilla(){
        return casilla;
    }
    
    public String getNombre() {
        return nombre;
    }

    Boolean getHipotecada() {
        return hipotecado;
    }

    int getAlquilerBase() {
        return alquilerBase;
    }

    double getFactorRevaloracion() {
        return factorRevaloracion;
    }

    int getHipotecaBase() {
        return hipotecaBase;
    }

    int getPrecioEdificar() {
        return precioEdificar;
    }

    void setHipotecada(Boolean hipotecado) {
        this.hipotecado = hipotecado;
    }
    
    Boolean propietarioEncacelado() {
        return propietario.getEncarcelado();
    }
    
    void cobrarAlquiler(int coste) {
        propietario.modificarSaldo(coste);
    }
    
    Boolean tengoPropietario() {
        return propietario != null;
    }
    
    void setCasilla(Casilla casilla) {
        this.casilla = casilla;
    }

    void setPropietario(Jugador propietario) {
        this.propietario = propietario;
    }
    
    @Override
    public String toString() {
        return "{ calle= " + nombre + ", hipotecado= " + hipotecado + ", alquilerBase= " + alquilerBase + ", factorRevaloracion= " + factorRevaloracion + ", hipotecaBase= " + hipotecaBase + ", precioEdificar= " + precioEdificar + '}';
    }
    
    
}
